components/profile_selector.js

```diff
@@ -3,75 +3,104 @@ import PropTypes from "prop-types";
-  showServiceType
-} from "../selectors/show_filters";
 
 export class ProfileSelector extends Component {
-  render() {
+  showQuestion = (question, index, reduxState) => {
     const {
-      t,
-      showServiceType,
-      showStatusAndVitals,
-      showServiceHealthIssue
-    } = this.props;
-    return (
-      <div>
-        <Grid container spacing={8}>
-          <Grid item xs={12} className="patronTypeFilter">
+      questions,
+      multipleChoiceOptions,
+      questionDisplayLogic
+    } = reduxState;
+
+    let questionDict = {};
+    questions.forEach(x => {
+      questionDict[x.id] = x.variable_name;
+    });
+
+    let optionDict = {};
+    multipleChoiceOptions.forEach(x => {
+      optionDict[x.variable_name] = x.id;
+    });
+
+    if (index === 0) {
+      return true;
+    }
+
+    // show if the previous question has an answer
+    const previousQuestionAnswered =
+      reduxState[questions[index - 1].variable_name] !== "";
+    if (!previousQuestionAnswered) {
+      return false;
+    }
+
+    const relevantLogic = questionDisplayLogic.filter(x => {
+      return (
+        x["exclude questions"] &&
+        x["exclude questions"].indexOf(question.id) > -1
+      );
+    });
+
+    if (!relevantLogic) {
+      return true;
+    }
+
+    let return_value = true;
+    relevantLogic.forEach(x => {
+      const questionId = x.question[0];
+      const users_answer = reduxState[questionDict[questionId]];
+      const users_answer_id = optionDict[users_answer];
+      if (x["has value"].indexOf(users_answer_id) > -1) {
+        return_value = false;
+      }
+    });
+
+    return return_value;
+  };
+
+  render() {
+    const { t } = this.props;
+    const { questions, multipleChoiceOptions } = this.props.reduxState;
+    let jsx_array = [];
+
+    questions.forEach((question, index) => {
+      if (this.showQuestion(question, index, this.props.reduxState)) {
+        const options = multipleChoiceOptions
+          .filter(mco => question.id === mco.linked_question[0])
+          .map(x => x.variable_name);
+
+        jsx_array.push(
+          <Grid
+            item
+            xs={12}
+            key={question.variable_name + "Filter"}
+            className={question.variable_name + "Filter"}
+          >
             <RadioSelector
               t={t}
-              legend={t("B3.Filter by eligibility")}
-              selectorType={"patronType"}
+              legend={
+                t("current-language-code") === "en"
+                  ? question.display_text_english
+                  : question.display_text_french
+              }
+              selectorType={question.variable_name}
+              options={options}
               store={this.props.store}
             />
           </Grid>
+        );
+      }
+    });
+
+    // legend={t(
+    //   this.props.statusAndVitals === "deceased"
+    //     ? "health issue question deceased"
+    //     : "health issue question"
+    // )}
```

Comment:
Let's 🔥 this commented code for now and add an issue to address the alive / deceased language choice issue.